src/biogeochem/CNNStateUpdate1Mod.F90

```diff
@@ -57,22 +61,36 @@ subroutine NStateUpdateDynPatch(bounds, num_soilc_with_inactive, filter_soilc_wi
     associate( &
          nf_veg => cnveg_nitrogenflux_inst  , &
          ns_veg => cnveg_nitrogenstate_inst , &
+         nf_soil => soilbiogeochem_nitrogenflux_inst, &
          ns_soil => soilbiogeochem_nitrogenstate_inst &
          )
 
     dt = get_step_size_real()
-
+    nstep = get_nstep()
     do j = 1, nlevdecomp
        do fc = 1, num_soilc_with_inactive
           c = filter_soilc_with_inactive(fc)
-          ns_soil%decomp_npools_vr_col(c,j,i_met_lit) = ns_soil%decomp_npools_vr_col(c,j,i_met_lit) + &
+    !      if (.not. use_soil_matrixcn) then
+             ns_soil%decomp_npools_vr_col(c,j,i_met_lit) = ns_soil%decomp_npools_vr_col(c,j,i_met_lit) + &
                nf_veg%dwt_frootn_to_litr_met_n_col(c,j) * dt
-          ns_soil%decomp_npools_vr_col(c,j,i_cel_lit) = ns_soil%decomp_npools_vr_col(c,j,i_cel_lit) + &
+             ns_soil%decomp_npools_vr_col(c,j,i_cel_lit) = ns_soil%decomp_npools_vr_col(c,j,i_cel_lit) + &
                nf_veg%dwt_frootn_to_litr_cel_n_col(c,j) * dt
-          ns_soil%decomp_npools_vr_col(c,j,i_lig_lit) = ns_soil%decomp_npools_vr_col(c,j,i_lig_lit) + &
+             ns_soil%decomp_npools_vr_col(c,j,i_lig_lit) = ns_soil%decomp_npools_vr_col(c,j,i_lig_lit) + &
                nf_veg%dwt_frootn_to_litr_lig_n_col(c,j) * dt
-          ns_soil%decomp_npools_vr_col(c,j,i_cwd) = ns_soil%decomp_npools_vr_col(c,j,i_cwd) + &
+             ns_soil%decomp_npools_vr_col(c,j,i_cwd) = ns_soil%decomp_npools_vr_col(c,j,i_cwd) + &
                ( nf_veg%dwt_livecrootn_to_cwdn_col(c,j) + nf_veg%dwt_deadcrootn_to_cwdn_col(c,j) ) * dt
+    !      else
```

Comment:
Please clean up your codes and commented out lines and subroutines from all your files.